instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/BaseTracer.java

```diff
@@ -81,14 +81,27 @@ public Span getCurrentSpan() {
   }
 
   protected final boolean shouldStartSpan(Kind proposedKind, Context context) {
+    boolean result;
     switch (proposedKind) {
       case CLIENT:
-        return !inClientSpan(context);
+        result = !inClientSpan(context);
+        break;
       case SERVER:
-        return !inServerSpan(context);
+        result = !inServerSpan(context);
+        break;
       default:
-        return true;
+        result = true;
```

Comment:
you could move this assignment to the declaration and remove the default to save a couple lines of code.

Reply:
done